src/main/kotlin/lotto/LottoApplication.kt

```diff
@@ -2,27 +2,42 @@ package lotto
 
 import lotto.model.DrawNumbers
 import lotto.model.Lotto
+import lotto.model.LottoGenerator
 import lotto.model.LottoNumber
-import lotto.model.RandomLottoGenerator
+import lotto.model.Lottos
+import lotto.model.RandomLottoGenerating
+import lotto.model.WinningStatistics
 import lotto.view.InputView
 import lotto.view.ResultView
 
 object LottoApplication {
-    fun run() {
-        InputView.printPaymentPriceInputMessage()
-        val paymentPrice = InputView.inputPaymentPrice()
+    fun run(inputView: InputView, resultView: ResultView) {
+        inputView.printPaymentPriceInputMessage()
+        val paymentPrice = inputView.inputPaymentPrice()
 
-        val purchasedLottos = RandomLottoGenerator.generateLottos(paymentPrice)
-        ResultView.printPurchasedLottoCount(purchasedLottos.size)
-        ResultView.printPurchasedLottos(purchasedLottos)
+        inputView.printCustomLottoCountInputMessage()
+        val customLottoCount = inputView.inputCustomLottoCount()
 
-        InputView.printWinningNumbersInputMessage()
-        val winningNumbers = Lotto(InputView.inputWinningNumbers())
+        inputView.printCustomLottoInputMessage()
+        val customLottos = mutableListOf<Lotto>()
+        repeat(customLottoCount) {
+            customLottos.add(inputView.inputCustomLotto())
+        }
 
-        InputView.printBonusNumberInputMessage()
-        val bonusNumber = LottoNumber.valueOf(InputView.inputBonusNumber())
+        val totalLottos = LottoGenerator.generateLottos(
+            paymentPrice, Lottos(customLottos),
+            RandomLottoGenerating
+        )
```

Comment:
named parameter와 trailing comma를 적극적으로 사용해주시는 것이 변경할때 추가할때 굉장히 좋습니다~!
https://kotlinlang.org/docs/coding-conventions.html#trailing-commas
```suggestion
        val totalLottos = LottoGenerator.generateLottos(
            paymentPrice = paymentPrice,
            customLottos = Lottos(customLottos),
            lottoGeneratingStrategy = RandomLottoGenerating,
        )
```